File: Selenium/Activities/Activity17.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity17 {
    public static void main(String[] args)
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver=new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Title: "+driver.getTitle());

        WebElement dropdown=driver.findElement(By.id("single-select"));
        Select option=new Select(dropdown);
        option.selectByVisibleText("Option 2");
        System.out.println("selection by visible text: "+option.getFirstSelectedOption().getText());
        option.selectByIndex(3);
        System.out.println("selection by Index: "+option.getFirstSelectedOption().getText());
        option.selectByValue("4");
        System.out.println("selection by Value: "+option.getFirstSelectedOption().getText());
        System.out.println("All options in the List:");
        List<WebElement> options=option.getOptions();
        for(WebElement ele:options) {
            System.out.println(ele.getText());
        }
        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.select import Select
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver:

    driver.get("https://v1.training-support.net/selenium/selects")
    print("Title: " + driver.title)

    dropdown=driver.find_element(By.ID,"single-select")
    option=Select(dropdown)
    option.select_by_visible_text("Option 2")
    print("selection by visible text: ",option.first_selected_option.text)
    option.select_by_index(3)
    print("selection by Index: ",option.first_selected_option.text)
    option.select_by_value("4")
    print("selection by Value: ",option.first_selected_option.text)
    print("All options in the List:")
    options=option.options
    for ele in options:
        print(ele.text)
    driver.quit()

